crates/server/src/store/metrics.rs

```diff
@@ -0,0 +1,69 @@
+use metrics::{Counter, Gauge, Unit, counter, describe_counter, describe_gauge, gauge};
+
+pub(super) struct PersistentStoreMetrics {
+    pub(super) zone_records_total: Gauge,
+    #[cfg(feature = "__dnssec")]
+    pub(super) zone_records_dynamically_updated: Counter,
+}
+
+pub(super) struct QueryStoreMetrics {
+    pub(super) zone_record_lookups: Counter,
+}
+
+pub(super) struct StoreMetrics {
+    pub(crate) query: QueryStoreMetrics,
+    pub(crate) persistent: PersistentStoreMetrics,
+}
+
+impl StoreMetrics {
+    pub(super) fn new(store: &'static str) -> Self {
+        Self {
+            query: QueryStoreMetrics::new(store),
+            persistent: PersistentStoreMetrics::new(store),
+        }
+    }
+}
+
+impl PersistentStoreMetrics {
```

Comment:
Nit: keep the impls below the type definitions, and sort as `StoreMetrics`, `PersistentStoreMetrics`, `QueryMetrics` (high complexity/depending to lower complexity/dependent).

Reply:
Reordered.